### PR TITLE
HHH-14835 : More changes to support SchemaMigrator/SchemaValidator using Hibernate Reactive

### PR DESCRIPTION
**Pull request by gbadner**
https://hibernate.atlassian.net/browse/HHH-14835

### CODE REVIEW

**Comment by gbadner** [comment]
Can this be changed to a draft?

**Comment by DavideD** [comment]
> DON'T MERGE YET; MORE CHANGES MAY BE NEEDED FOR DB2.

Couldn't we apply the changes needed for Db2 as a separate pr?

**Comment by gbadner** [comment]
> 
> Couldn't we apply the changes needed for Db2 as a separate pr?

@DavideD, yes that would be fine. DB2 should be very close. I just need to review it. I just thought it might be worth waiting a (hopefully) a day or two, then this issue could cover DB2 as well.

I also had a very small fix related to Sybase that Andrea mentioned got fixed in 6, that doesn't really warrant a separate issue.

I'm fine with creating a separate issue for DB2 though. Whatever works.

**Comment by DavideD** [comment]
> I'm fine with creating a separate issue for DB2 though. Whatever works.

We should check when the next ORM Final release is planned (@Sanne? or @dreab8?). If the MSSQL part is ready, it would be nice to have it working in the final release of Hibernate Reactive, even if the Db2 part is missing.

But if we are close to have both, a single PR sounds good.

**Comment by gbadner** [comment]
Force-pushed with amended commit that fixes an issue @dreab8 found on Sybase that turned up after merging with wip/6.0 branch.

**Comment by gbadner** [comment]
I started looking at DB2 work, but now I'm seeing failures due to the Vert.X bug with both `INDIVIDUALLY` and `GROUPED` strategies.

It looks like we're going to have to wait for the Vert.X bug to be fixed before being able to support `SchemaMigrator` on DB2.

Provided all the checks pass, I think it's fine to merge this PR.

File: hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaExportSqlServerWithSequenceDefaultSchemaCatalog.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author Andrea Boriero
+ */
+@TestForIssue(jiraKey = "HHH-XXX")
```
Comment (Sanne) [inline hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaExportSqlServerWithSequenceDefaultSchemaCatalog.java:38]: Could you fix the `jiraKey` - and  I guess the author, or did Andrea help here?

Comment (gbadner) [reply hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaExportSqlServerWithSequenceDefaultSchemaCatalog.java:38]: @Sanne, there is no jira key. This was cherry-picked from https://github.com/hibernate/hibernate-orm/commit/aaba4767fe2842c442104dff0a377a0f63ecffdd .

@Andrea and I agreed that it was not necessary to create a new jira issue for this.

I suppose I could fill in "HHH-14835" for the jira key though.

Comment (gbadner) [reply hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaExportSqlServerWithSequenceDefaultSchemaCatalog.java:38]: Oops, sorry, my mistake. That was not from cherry-picking. I'll fix it...

Comment (gbadner) [reply hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaExportSqlServerWithSequenceDefaultSchemaCatalog.java]: Fixed in https://github.com/hibernate/hibernate-orm/pull/4235/commits/823be540cae36c39b2de0e5f3947f2a246984097 .

Comment (gbadner) [reply hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaExportSqlServerWithSequenceDefaultSchemaCatalog.java]: I copied the test from a test that Andrea wrote, with minor changes.

**Review by Sanne — APPROVED** [review]

**Comment by Sanne** [comment]
Thanks! merged it

**Pull request merged**